### PR TITLE
Chatapp

### PR DESCRIPTION
**Pull request by PatilAneesh**

### CODE REVIEW

File: src/modals/UserModals.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ export interface UserModal {
   _id: string;
   status: Status;
   newMessages: { [key: string]: string[] };
+  members: [{ name: string; id: string }];
```
Comment (ResourceOne2011) [inline src/modals/UserModals.tsx:40]: create a seperate interface for members abd assign that type to members

File: src/pages/chat/chatbox/Footer.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { Socket } from "socket.io-client";
 import httpMethods from "../../../api/Service";
 import { FileModel } from "../../../modals/MessageModals";
 import { ClientModal } from "../../../modals/ClientModals";
+import { GroupInterface } from "../sidebar/Groups";
```
Comment (ResourceOne2011) [inline src/pages/chat/chatbox/Footer.tsx:13]: Remove the unused variables

File: src/pages/chat/sidebar/CreateGroupModal.tsx [hunk 3]
```diff
@@ -14,6 +14,7 @@ interface CreateGroupModel {
   name: string;
   members: { name: string; id: string }[];
   description: string;
+  admin: { name: string; id: string };
```
Comment (ResourceOne2011) [inline src/pages/chat/sidebar/CreateGroupModal.tsx:17]: create a new interface for { name: string; id: string };  in the global and assign wherever its needed

File: src/pages/chat/sidebar/Groups.tsx [hunk 4]
```diff
@@ -6,6 +6,8 @@ import { getFullName } from "../../../utils/utils";
 import { Button } from "react-bootstrap";
 import ReusableModal from "../../../utils/modal/ReusableModal";
 import CreateGroup from "./CreateGroupModal";
+import { Socket } from "socket.io-client";
+import { error } from "console";
```
Comment (ResourceOne2011) [inline src/pages/chat/sidebar/Groups.tsx:10]: is there any use of error import from console

File: src/modals/UserModals.tsx (hunk 1 above)
Comment (PatilAneesh) [reply src/modals/UserModals.tsx:40]: done

File: src/pages/chat/chatbox/Footer.tsx (hunk 2 above)
Comment (PatilAneesh) [reply src/pages/chat/chatbox/Footer.tsx:13]: done

File: src/pages/chat/sidebar/CreateGroupModal.tsx (hunk 3 above)
Comment (PatilAneesh) [reply src/pages/chat/sidebar/CreateGroupModal.tsx:17]: done

File: src/pages/chat/sidebar/Groups.tsx (hunk 4 above)
Comment (PatilAneesh) [reply src/pages/chat/sidebar/Groups.tsx:10]: done

File: src/modals/UserModals.tsx [hunk 5]
```diff
@@ -13,6 +13,8 @@ export interface UserContext {
   setCurrentRoom: React.Dispatch<React.SetStateAction<string>>;
 }
 export type Status = "Available" | "Break" | "Offline" | "On Ticket";
+
+export type Members = [{ name: string; id: string }];
```
Comment (ResourceOne2011) [inline src/modals/UserModals.tsx:17]: FIX

File: src/pages/chat/sidebar/CreateGroupModal.tsx [hunk 6]
```diff
@@ -10,10 +10,15 @@ import { GroupInterface } from "./Groups";
 interface CreateGroupProps {
   onCreateGroup: (group: GroupInterface) => void;
 }
+interface NameIdInterface {
```
Comment (ResourceOne2011) [inline src/pages/chat/sidebar/CreateGroupModal.tsx:13]: Create only one and import

**Review by ResourceOne2011 — APPROVED** [review]
OK

**Pull request merged**